### PR TITLE
Initial notebook for promptflow based simulator

### PR DESCRIPTION
**Pull request by nagkumar91**
# Description

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure-Samples/azureai-samples/blob/main/CONTRIBUTING.md)
- [ ] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure-Samples/azureai-samples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -162,3 +162,8 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+# macOS Finder
+.DS_Store
+*.ignore.*
+*outputs*
```
Comment (kdestin) [inline .gitignore:169]: This feels like an overly broad inclusion into the root gitignore. 

If this is exclusion is important to your additions, consider adding a .gitignore with these lines closer to the code that creates these files

File: scenarios/evaluate-an-endpoint/evaluate-target.ipynb
Comment (kdestin) [inline scenarios/evaluate-an-endpoint/evaluate-target.ipynb:1]: Doesn't adhere to notebook template: https://github.com/Azure-Samples/azureai-samples/blob/9154f3b2ad3b4250da0b03e84d69edac56d6cc22/scenarios/template.ipynb

File: scenarios/generate-synthetic-data/simulate-adversarial-interactions/askwiki/simulate_and_evaluate_ask_wiki.ipynb [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+   "source": [
+    "# Adversarial Simulator for a custom application - askwiki\n",
+    "\n",
+    "## Objective\n",
+    "\n",
+    "This tutorial provides a step-by-step guide on how to leverage adversarial simulator to simulate an adversarial question answering scenario against a custom application - askwiki.\n",
+    "\n",
+    "This tutorial uses the following Azure AI services:\n",
+    "\n",
+    "- promptflow-evals\n",
```
Comment (kdestin) [inline scenarios/generate-synthetic-data/simulate-adversarial-interactions/askwiki/simulate_and_evaluate_ask_wiki.ipynb:15]: Consider adding a link to relevant documentation

File: scenarios/generate-synthetic-data/simulate-adversarial-interactions/promptflow-online-endpoint/simulate_and_evaluate_online_endpoint.ipynb [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+   "source": [
+    "# Adversarial Simulator for an online endpont\n",
+    "\n",
+    "## Objective\n",
+    "\n",
+    "This tutorial provides a step-by-step guide on how to leverage adversarial simulator to simulate an adversarial question answering scenario against an online endpoint\n",
+    "\n",
+    "This tutorial uses the following Azure AI services:\n",
+    "\n",
+    "- promptflow-evals\n",
```
Comment (kdestin) [inline scenarios/generate-synthetic-data/simulate-adversarial-interactions/promptflow-online-endpoint/simulate_and_evaluate_online_endpoint.ipynb:15]: Consider adding a link to documentation

File: scenarios/generate-synthetic-data/simulate-adversarial-interactions/rag/simulate_and_evaluate_rag.ipynb [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+    "# Adversarial Simulator for a custom rag application\n",
+    "\n",
+    "## Objective\n",
+    "\n",
+    "This tutorial provides a step-by-step guide on how to leverage adversarial simulator to simulate an adversarial question answering scenario against a custom rag application.\n",
+    "\n",
+    "This tutorial uses the following Azure AI services:\n",
+    "\n",
+    "- promptflow-evals\n",
+    "- promptflow-rag\n",
```
Comment (kdestin) [inline scenarios/generate-synthetic-data/simulate-adversarial-interactions/rag/simulate_and_evaluate_rag.ipynb:16]: Consider linking to the relevant documentation

**Review by kdestin — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (nagkumar91) [reply .gitignore:169]: yeah, makes sense.

**Review by diondrapeck — APPROVED** [review]

File: scenarios/evaluate-an-endpoint/evaluate-target.ipynb
Comment (nagkumar91) [reply scenarios/evaluate-an-endpoint/evaluate-target.ipynb:1]: Fixed it :)

File: scenarios/generate-synthetic-data/simulate-adversarial-interactions/askwiki/simulate_and_evaluate_ask_wiki.ipynb (hunk 2 above)
Comment (nagkumar91) [reply scenarios/generate-synthetic-data/simulate-adversarial-interactions/askwiki/simulate_and_evaluate_ask_wiki.ipynb]: Added

File: scenarios/generate-synthetic-data/simulate-adversarial-interactions/promptflow-online-endpoint/simulate_and_evaluate_online_endpoint.ipynb (hunk 3 above)
Comment (nagkumar91) [reply scenarios/generate-synthetic-data/simulate-adversarial-interactions/promptflow-online-endpoint/simulate_and_evaluate_online_endpoint.ipynb]: Done :)

File: scenarios/generate-synthetic-data/simulate-adversarial-interactions/rag/simulate_and_evaluate_rag.ipynb (hunk 4 above)
Comment (nagkumar91) [reply scenarios/generate-synthetic-data/simulate-adversarial-interactions/rag/simulate_and_evaluate_rag.ipynb]: RAG sample has been removed and will be re-added in a new PR as the `promptflow-rag` team is still working on making the cloud index work.

**Review by kdestin — APPROVED** [review]

**Pull request merged**